### ISSUE TITLE
Verbose indicator: show keymap name/description for prefix keymaps

### ISSUE THREAD

**Issue by minad** [oantolin/embark#306]
Afaik the name is available for keymaps set via fset.

**Comment by oantolin** [comment]
I don't understand what you mean about fset, but just like simple menu items, a keymap can include a string which is used a description for the whole menu. We could use that.

**Comment by minad** [comment]
There is some magic.

~~~ elisp
(global-set-key (kbd "C-c X") 'my-keymap)
(fset 'my-keymap
      (let ((map (make-sparse-keymap)))
        (define-key map "X" (lambda () (interactive) (message "Test")))
        map))
~~~

**Comment by oantolin** [comment]
Oh, but given the keymap, how do you recover the symbol? Also, I most keymaps are not defined like that, but just stored in some variable.

**Comment by oantolin** [comment]
Maybe iterating over the obarray looking for the keymap is not too bad.

**Comment by minad** [comment]
> Oh, but given the keymap, how do you recover the symbol? Also, I most keymaps are not defined like that, but just stored in some variable.

During recursion over the keymap we can obtain the symbol. I think many keymaps are defined like this actually, since it also helps which-key. And you can easily fix it for the ones which are not defined like this with fset and changing the prefix keybinding.

Iterating over the obarray is another solution of course, ideally in combination with a cache.

**Comment by bdarcus** [comment]
> I think many keymaps are defined like this actually, since it also helps which-key.

You mean like this?

https://github.com/oantolin/embark/issues/306#issuecomment-889410239

Is that preferred, as in I should do that?

**Comment by minad** [comment]
@bdarcus Yes, this is the preferred way to name keymaps as far as I understand.

**Comment by minad** [comment]
Fixed in #306.
